### PR TITLE
Recipe/119 preprocessor usage and performance eval

### PR DESCRIPTION
**Pull request by fayvor**
Per [pm/#119](https://github.com/ibm-granite-community/pm/issues/119):
* Use Preprocessor
* Split the data
* Evaluate model performance
* Use trained model for forward predictions

### CODE REVIEW

**Comment by fayvor** [comment]
@wgifford please take a look at this draft and let me know what you think.

In particular, when I go to make a forward prediction using the trained and evaluated model, am I doing that correctly? I assume the model gets modified by the Trainer; is that correct? Is the original model object modified, or does the Trainer make a copy? Ie., do I have to use `trainer.model` for the forward prediction?

**Comment by wgifford** [comment]
I was thinking the following:
- remove `encode_categorical=False,` from TSP
- instead of "get_datasets" which return torch datasets, use: 
```
train_df, valid_df, test_df = prepare_data_splits(
    input_df,
    context_length=params["context_length"],
    split_config=split_params,
)
```
this will give you dataframes.
- do not use the zero-shot trainer and/or their evaluate method, go directly to the forecasting pipeline using the test data
- you will need to train tsp on train_df, then pass tsp.preprocess(test_df) to the forecasting pipeline
- don't trim the test data to context length, or use the explode option in the pipeline (we want all the predictions for the test data)
- for plotting, plot a few chosen examples (using indices) with the ground truth showing
- add a separate evaluation function to show how a custom metric can be implemented (I have an example I can provide).

**Comment by fayvor** [comment]
@wgifford I've made those changes. LMK what you think.

A few things:
1. The PvA plot values are in scaled units. How can I convert back to the original units of the incoming data? Do you have a "de-scaler" Postprocessor?
2. I'd like to include forward prediction with pre-processed data, since it is what many users will do in production after evaluating the model. I'll need the "de-scaler" for this.
3. I'll need the evaluation function from you when you have it.

**Comment by fayvor** [comment]
@wgifford, I merged your commit into my branch, but it's not signed. Could you amend the commit so that it is both "signed" (GPG/SSH) and "signed-off" (DCO)? 

The instructions are in the first two bullets [here](https://github.com/ibm-granite-community/granite-timeseries-cookbook/tree/main?tab=readme-ov-file#other-general-guidance):
- The Developer Certificate of Origin (DCO) applies to the code, documentation, and any example data provided. See community CONTRIBUTING guide [legal section](https://github.com/ibm-granite-cookbooks/community/blob/main/CONTRIBUTING.md#legal) for more on the DCO.
- Commits must be GPG or SSH signed. See the community CONTRIBUTING guide [signing commits section](https://github.com/ibm-granite-cookbooks/community/blob/main/CONTRIBUTING.md#signing-commits) for details.

File: recipes/Time_Series/Preprocessor_Use_and_Performance_Evaluation.ipynb [hunk 1]
```diff
@@ -870,12 +870,20 @@
...
    "metadata": {},
    "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/Users/wmgifford/miniconda3/envs/granite-tsfm/lib/python3.12/site-packages/pandas/core/indexes/base.py:7588: FutureWarning: Dtype inference on a pandas object (Series, Index, ExtensionArray) is deprecated. The Index constructor will keep the original dtype in the future. Call `infer_objects` on the result to get the old behavior.\n",
+      "  return Index(sequences[0], name=names)\n"
+     ]
+    },
```
Comment (fayvor) [inline recipes/Time_Series/Preprocessor_Use_and_Performance_Evaluation.ipynb:883]: @wgifford let's keep these warnings out if possible.

Comment (wgifford) [reply recipes/Time_Series/Preprocessor_Use_and_Performance_Evaluation.ipynb]: Yes, will look into it.

**Comment by fayvor** [comment]
@wgifford thanks for this! Taking a look.

**Comment by fayvor** [comment]
Made some minor changes. I will go ahead and merge.

**Comment by wgifford** [comment]
I am not a huge fan of casting the dates, but if we must avoid the warning, I don't really see an alternative.

**Pull request merged**